### ISSUE TITLE
Organize Favorites section might have glitches

### ISSUE THREAD

**Issue by xLn2** [clsid2/mpc-hc#1574]
> Mpc-hc 1.9.9.34 - Windows 10 21h1

Timestamp rendered weird (same behavior when renaming item):
https://i.imgur.com/VWvGEFS.gif

Favorite items stand out on hover.
https://i.imgur.com/Lv9pL2r.gif
on item hover it gives a space on text. It might be intended behavior but all items which mouse hover effected also stand out. I think only latest one should affected. 
one example 
![image](https://user-images.githubusercontent.com/3073922/155926465-8d5b2d52-6f96-4db7-b111-cb4b670d6b0b.png)

Renaming render issue (adipose did something on playlist renaming might applied on this) :
https://i.imgur.com/PJR4ePz.gif

When resized window timestamps slide.
https://i.imgur.com/yZenzbZ.gif

Tooltip does not show file path in some cases.  I did not know how to trigger. (Same behavior seems on playlist section.)
https://i.imgur.com/9HFjfxp.gif

I think organize favorites part of mpc-hc is very old. Since its not very well used by most people, it does not take enough attention.

**Comment by xLn2** [comment]
I have found that white theme has less glitches. There is no standing out on item hover and Tooltip shows file path correctly in white theme. Position of the timestamp is depends on title of video's, so it probably renders only with rename and File tab window reinitialized. Mpc-Be is used align right method and always rendered while window resized, it might help. Mpc-Be also same rename behavior (ghost old text). 
Mpc-Be has good features like edit, you can edit title and file path. I have suggestion to interested developers to add time edit and if there is A-B loop also edit these settings can help to users.
![image](https://user-images.githubusercontent.com/3073922/156592225-eacea76a-c68f-4816-b6d7-cd9e6fe42b9b.png)

**Comment by adipose** [comment]
They are both glitchy on the rename redraw.  Also the timestamps jump around in white mode.

I see the background redraw. Not sure why it's behaving that way.  The previous fix should have helped...

**Comment by adipose** [comment]
https://github.com/clsid2/mpc-hc/pull/1632

This had a custom draw routine, which used Textout and some other weird custom spacing.  I replaced it with some smarter draw code that works better, also put in some logic to handle the drawing under edit control.

For some reason the playlist has fewer issues than this dialog with the edit control, but I am not sure why.  But try it out when there's a build out.

**Comment by xLn2** [comment]
I tested it with 1.9.20.24.
It has very clean view right now :) timestamps are not visible
![resim](https://user-images.githubusercontent.com/3073922/160894625-bd180c5e-3c89-46e6-9d5a-f0c2541b0d17.png)

Timestamps show up on hover where timestamp actually should have been:
![resim](https://user-images.githubusercontent.com/3073922/160898646-dd1d76d9-fa33-4902-b512-cb504d05b38b.png)

Timestamps appear on hover but disappear with click:
https://i.imgur.com/GilhFdq.gif

Timestamps tries to appear:
https://i.imgur.com/qBSuBjd.gif

standing out and rename behavior looks like fixed.

**Comment by adipose** [comment]
I have been working on it some more.  I found several issues:

1. On edit, the column resize algorithm is fired.  It sets the timestamp column to autosize, then uses this to calculate the "real" size, and sets it.  But somehow, the autosize algorithm triggers a paint of both cases, even if painting is disabled while updates are done.  I have changed it to calculate the min size only once, which prevents the flashing/moving column.

2. There was a bug in the custom listctrl that allowed double draw code--one from the listctrl custom code, and one from the "ownerdraw" code specific to certain lists, like the favorite organizing one.  I confirmed it also affected the playlist, but for whatever reason, that particular ownerdraw code did not cause any visible glitches.  In particular, the custom draw code could be called when doing a hover event, like the tooltip.  On a normal draw, the customdraw code didn't get called, but in special cases, it did.  So now, I detect the presence of ownerdraw styles, and avoid the customdraw step that duplicates it.  One interesting effect is the ownerdraw is unable to draw the far left pixel of the highlight, but customdraw is able to.  Perhaps a "glitch" in Microsoft's implementations...

3. Timestamp was moving around during hover, I believe related to # 2.  It depends on the alignment of that column how far it moves around.  But now, it should be handled by the ownerdraw code.

https://github.com/clsid2/mpc-hc/pull/1635

**Comment by xLn2** [comment]
changes with 1.9.20.46 , organize favorites edit page buttons looks invalid.
![resim](https://user-images.githubusercontent.com/3073922/162544115-633573e7-2cde-4513-8b5e-d2c51342cda3.png)


in 1.9.20.36

![resim](https://user-images.githubusercontent.com/3073922/162543983-ef708c83-fff6-4f01-9bf6-ac2afdbee740.png)

I dont know if it is important but both versions have bottom scroll shows up a short amount of time.
![resim](https://user-images.githubusercontent.com/3073922/162544230-90af416c-7de2-474d-98f4-8017a244919d.png)

Also in 1.9.20.46 when you add favorites a item it shows at favorities menu bar but does not seem in organize fav window until you restart the app.

**Comment by adipose** [comment]
Thanks for feedback.

1. Scrollbar isn't meant to show but I could disable entirely to prevent glitch.

2. Something strange with buttons. I will check.

3. Refresh on adding favorites wasn't required when dialog was modal. Now I will have to pass a message to update. New feature for modeless.

**Comment by xLn2** [comment]
in 1.9.20.56,
when you delete items on organize favorite window, click OK,
items are still visible on menu which are supposed to be deleted, 
to be sure open organize favorites window and you see items are look like deleted,
add one item to favorites and you will see all items which are supposed to be deleted, come back to organize favorites window again.
actually none of them able to deleted, even you close and open mpchc you can see them

**Comment by adipose** [comment]
https://github.com/clsid2/mpc-hc/pull/1669

My mistake.  The list positions were getting invalidated, causing tooltip crashes.  I removed the code that cleared them during save, thinking it was unnecessary since the dialog never gets destroyed.  But now I see, it was actually updating the list for later persisting.  In order to load valid positions, I now reload the list after the save.

**Comment by adipose** [comment]
Re: the horizontal scrollbar, I am going to leave it alone for now. By adding a very long filename, I was able to get it to widen enough to need scrolling.  Also just resizing the form can cause it to need a scrollbar.  So rather than hide it initially, I'll leave it there.

I haven't been able to observe it actually turning on and then off as you described.  I'm sure it has to do with the column resizing that happens initially.  That code could be re-written to hard code the last column width, probably.  If it really is bugging you go ahead and open another issue and upload a video of it happening.

**Comment by xLn2** [comment]
Actually Its not important stuff and it only appears when swapping between dvd and file tab. Only flashed for just around 0.5sec. I did report horizontal scroll because of I assume swapping file and dvd tab has same behavior with opening organize favorites window from menu. I was wrong about that and probably I lead misunderstanding and took your time. In case curiosity about horizontal scrool flashing I added gif below and if it is important, i will open an issue.
swapping DVD tab to File tab in Organize Favorites window:
https://i.imgur.com/gkqhFNr.gif

**Comment by adipose** [comment]
Oh, I can replicate it now.  Yeah I will take a look.

**Comment by adipose** [comment]
https://github.com/clsid2/mpc-hc/pull/1671

So it's a simple redraw problem.  When switching tabs it completely reloads the list.  That causes it to change size a bit until it's reloaded.  Seems pretty inefficient code to me, but as long as I don't draw while it's going, it's ok.

I assume the reason it's there is in case one tab is first opened, the other isn't loaded yet, so it makes sure when switching.  But the result is, it will lose what you were working on when you switch.  Seems like loading them all in the beginning is all that would be needed.

Anyway, at least the draw is better now.
